frontend/packages/dev-console/src/components/helm/HelmInstallUpgradePage.tsx

```diff
@@ -0,0 +1,183 @@
+import * as React from 'react';
+import * as _ from 'lodash';
+import { safeDump, safeLoad } from 'js-yaml';
+import { Formik } from 'formik';
+import { Helmet } from 'react-helmet';
+import { RouteComponentProps } from 'react-router-dom';
+import { PageHeading, history, LoadingBox } from '@console/internal/components/utils';
+import { coFetchJSON } from '@console/internal/co-fetch';
+import { PageBody } from '@console/shared';
+import NamespacedPage, { NamespacedPageVariants } from '../NamespacedPage';
+import HelmInstallUpgradeForm from './HelmInstallUpgradeForm';
+import { HelmActionType, HelmChart, HelmActionConfigType } from './helm-types';
+import { getValidationSchema } from './helm-validation-utils';
+
+export type HelmInstallUpgradePageProps = RouteComponentProps<{
+  ns?: string;
+  releaseName?: string;
+}>;
+
+export type HelmInstallUpgradeFormData = {
+  helmReleaseName: string;
+  helmChartURL?: string;
+  chartValuesYAML: string;
+  chartVersion?: string;
+};
+
+const HelmInstallUpgradePage: React.FunctionComponent<HelmInstallUpgradePageProps> = ({
+  location,
+  match,
+}) => {
+  const searchParams = new URLSearchParams(location.search);
+
+  const chartURL = decodeURIComponent(searchParams.get('chartURL'));
+  const namespace = match.params.ns || searchParams.get('preselected-ns');
+  const releaseName = !_.isEmpty(match.params.releaseName) && match.params.releaseName;
+  const helmChartName = searchParams.get('chartName');
+  const [chartDataLoaded, setChartDataLoaded] = React.useState(false);
+  const [chartName, setChartName] = React.useState('');
+  const [chartHasValues, setChartHasValues] = React.useState(false);
+  const [YAMLData, setYAMLData] = React.useState({});
+  const [activeChartVersion, setActiveChartVersion] = React.useState<string>('');
+  const helmAction: HelmActionType =
+    chartURL !== 'null' ? HelmActionType.Install : HelmActionType.Upgrade;
+
+  const HelmActionConfig: { [helmAction: string]: HelmActionConfigType } = {
+    Install: {
+      title: 'Install Helm Chart',
+      helmReleaseApi: `/api/helm/chart?url=${chartURL}`,
+      fetch: coFetchJSON.post,
+      redirectURL: `/topology/ns/${namespace}`,
+    },
+    Upgrade: {
+      title: 'Upgrade Helm Release',
+      helmReleaseApi: `/api/helm/release?ns=${namespace}&release_name=${releaseName}`,
+      fetch: coFetchJSON.put,
+      redirectURL: `/helm-releases/ns/${namespace}`,
+    },
+  };
+
```

Comment:
Why not just add `const config = HelmActionConfig[helmAction];` and then use `config.something` everywhere without having to do the boolean check again and again?